nano/node/node.cpp

```diff
@@ -520,6 +520,8 @@ void nano::node::keepalive (std::string const & address_a, uint16_t port_a)
 
 void nano::node::inbound (const nano::message & message, const std::shared_ptr<nano::transport::channel> & channel)
 {
+	debug_assert (channel->owner () == shared_from_this ()); // This node should be the channel owner
```

Comment:
We could have owner return a naked pointer and compare against 'this' since we're not intending to manage lifetime.  I'm fine with keeping this though since it's just test code.